### PR TITLE
Update openpgp.d.ts

### PR DESCRIPTION
**Pull request by sgnix**
Add missing interfaces and enum types

### CODE REVIEW

**Comment by vvakame** [comment]
@glacasa  could you review this PR?
@naturalist please fix travis ci failed. try with `--noImplicitAny`.

File: openpgp/openpgp.d.ts [hunk 1]
```diff
@@ -377,7 +384,18 @@ declare module openpgp.key {
      */
     interface Key {
         armor(): String,
-        decrypt(passphrase: String): Boolean,
+        decrypt(passphrase: string): boolean;
+        getExpirationTime(): Date;
+        getKeyIds(): Array<any>;
```
Comment (glacasa) [inline openpgp/openpgp.d.ts]: According to official doc, it should return an array of Keyid :
 getKeyIds(): Array<Keyid>;

File: openpgp/openpgp.d.ts [hunk 2]
```diff
@@ -377,7 +384,18 @@ declare module openpgp.key {
...
+        getPreferredHashAlgorithm(): string;
+        getPrimaryUser(): any;
+        getUserIds(): Array<string>;
+        isPrivate(): boolean;
+        isPublic(): boolean;
+        primaryKey: packet.PublicKey;
```
Comment (glacasa) [inline openpgp/openpgp.d.ts]: I can't find the primaryKey property in the doc

Comment (naturalist) [inline openpgp/openpgp.d.ts]: You have probably noticed that openpgp's documentations generally sucks. The primaryKey property is instrumental, yet they failed to add any documentation on it.

File: openpgp/openpgp.d.ts (hunk 1 above)
Comment (naturalist) [inline openpgp/openpgp.d.ts]: That's correct. I can change that, or you can - whichever you prefer.

**Comment by naturalist** [comment]
Fixed travis-ci and added the missing return type pointed by @glacasa. Closing this PR and will submit a new one in a second.

**Pull request closed without merging**